### ISSUE TITLE
Sometimes instanceof() function returns "[[Method]]" for a structure (YYC only)

### ISSUE THREAD

**Issue by phyrexiankid** [YoYoGames/GameMaker-Bugs#3929]
### Description

On YYC, the function `instanceof()` with struct literal as an argument returns string `[[Method]]` instead of `struct`, as expected due to [documentation ](https://manual.gamemaker.io/monthly/en/#t=GameMaker_Language%2FGML_Reference%2FVariable_Functions%2Finstanceof.htm&rhsearch=instanceof&rhhlterm=instanceo):
> If you pass the function a struct literal then it will simply return the string `"struct"`,

### Expected Change

Function `instanceof()` should always return `struct` for struct literals

### Steps To Reproduce

Empty project, one room and one object, object's Main event.
```gml
var thing = {a: 10, b: "abc"};
show_debug_message(instanceof(thing)); // Result: "[[Method]]"
var thing2 = {c: 11, d: "def"};
show_debug_message(instanceof(thing2)); // Result: "struct"
```

The easiest way to reproduce is to put it in the object's event (tested on Create and Step events). The unexpected result appears sometimes on first and sometimes on second instanceof(), and won't appear any more in the current event, doesn't matter how much times `instanceof()` was called.
On a clear project the issue only happens when instanceof() is called from object's event (and not from a function or method), howether on a bigger project is looks like it something happens in methods too. 

Reproducing on 2023.11 and 2023.1100.0.450 BETA versions.

Including the Contact Us Package and a sample project:
[instanceofbug.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13877212/instanceofbug.zip)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.11.1 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.2 - should be in Beta 2

**Comment by Gamer-XP** [comment]
@rwkay I got a similar weird issue with "instanceof" in the older GM version, but I can't check it with latest version because newer GM IDE does not start at all for some reason.

My issue is: when compiling with YYC, "instanceof" often returns string "Object" instead of correct constructor name. I'm using runtime 2023.1.1.81. Was it fixed, because I can't find instanceof being mentioned anywhere in release notes?

[InstanceOfBug.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13979298/InstanceOfBug.zip)

In this small project "instanceof" returns "Object" for me.

Can somebody confirm if it's still a bug?
